File: BackEndApplication/Models/Transaction.cs
using System;

namespace BackEndApplication.Models
{
    public class Transaction
    {
        public Guid Id { get; set; }

        public Guid AccountId { get; set; }

        public DateTime CreateDate { get; set; }

        public TransactionType Type { get; set; }

        public decimal Amount { get; set; }
    }
}

File: FrontEndApplication/src/models/transaction.ts
import { TransactionType } from './transaction-type';

export default class Transaction {
    id = "";
    accountId = "";
    createDate = new Date();
    type: TransactionType = TransactionType.Empty;
    amount = 0;

    static parseJson(json: any) {
        if (json === null || json === undefined) return json;

        const { createDate, ...rest } = json;

        return Object.assign(
            new Transaction(),
            {
                createDate: new Date(createDate)
            },
            rest);
    }
}